includes/Admin/Cross_Origin_Isolation.php

```diff
@@ -0,0 +1,308 @@
+		if ( ! $this->is_edit_screen() ) {
+			return $tag;
+		}
+
+		return $this->add_attribute( $tag, 'href', $href );
+	}
+
+	/**
+	 * Filters the HTML script tag of an enqueued script.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param string $tag    The `<script>` tag for the enqueued script.
+	 * @param string $handle The script's registered handle.
+	 * @param string $src    The script's source URL.
+	 *
+	 * @return string
+	 */
+	public function script_loader_tag( $tag, $handle, $src ) {
+		if ( ! $this->is_edit_screen() ) {
+			return $tag;
+		}
+
+		return $this->add_attribute( $tag, 'src', $src );
+	}
+
+	/**
+	 * Filter the avatar tag.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param string $avatar      HTML for the user's avatar.
+	 * @param mixed  $id_or_email The avatar to retrieve. Accepts a user_id, Gravatar MD5 hash,
+	 *                            user email, WP_User object, WP_Post object, or WP_Comment object.
+	 * @param int    $size        Square avatar width and height in pixels to retrieve.
+	 * @param string $default     URL for the default image or a default type. Accepts '404', 'retro', 'monsterid',
+	 *                            'wavatar', 'indenticon', 'mystery', 'mm', 'mysteryman', 'blank', or
+	 *                            'gravatar_default'. Default is the value of the 'avatar_default' option, with a
+	 *                            fallback of 'mystery'.
+	 * @param string $alt         Alternative text to use in the avatar image tag. Default empty.
+	 * @param array  $args        Arguments passed to get_avatar_data(), after processing.
+	 *
+	 * @return string
+	 */
+	public function get_avatar( $avatar, $id_or_email, $size, $default, $alt, $args ) {
+		if ( ! $this->is_edit_screen() ) {
+			return $avatar;
+		}
+
+		return $this->add_attribute( $avatar, 'src', $args['url'] );
+	}
+
+	/**
+	 * Do replacement to add crossorigin attribute.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param string $html HTML string.
+	 * @param string $attribute Attribute to check for.
+	 * @param string $url URL.
+	 *
+	 * @return string
+	 */
+	protected function add_attribute( $html, $attribute, $url ) {
+		$site_url = site_url();
+		$url      = esc_url( $url );
+
+		if ( $this->starts_with( $url, $site_url ) ) {
+			return $html;
+		}
+
+		if ( $this->starts_with( $url, '/' ) ) {
+			return $html;
+		}
+
+		return (string) str_replace( "{$attribute}='{$url}'", "crossorigin='anonymous' {$attribute}='{$url}'", $html );
+	}
+
+	/**
+	 * Does string start with.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @param string $string       String to search.
+	 * @param string $start_string String to search with.
+	 *
+	 * @return bool
+	 */
+	private function starts_with( $string, $start_string ) {
+		$len = strlen( $start_string );
+
+		return ( substr( $string, 0, $len ) === $start_string );
+	}
+
+	/**
+	 * Is this the editor screen.
+	 *
+	 * @since 1.6.0
+	 *
+	 * @return bool
+	 */
+	protected function is_edit_screen() {
```

Comment:
Can this be combined with `is_needed` into one method? Don't see why we need both.

Reply:
I think for unit testing reasons, they should be kept apart.

Reply:
At least now the `is_needed()` check is only in one place now 👍